### PR TITLE
MBL-1573: Checkout Screen (Change Reward flow for crowdfund) 

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What

New checkout screen using late pledges UI + entirely new VM
⚠️ Some UI small pieces will come
⚠️ Still work in progress more PR's will come with bunch of testing for the new VM!

# 👀 See

- Modify my current selection of Shipping location/Bonus/AddOns or payment method
https://github.com/user-attachments/assets/942bf183-f90f-4592-9271-c58e63fe1c92

- Select other reward
https://github.com/user-attachments/assets/e3d4cf6d-0ce2-4bc7-ac42-e0782eb917fd

|  |  |

# 📋 QA
- Use any project in staging that is still in campaign and not completed, [here one example](https://staging.kickstarter.com/projects/1521513553/a-really-cool-proj?ref=nav_search&result=project&term=a%20really%20cool%20proj)

# Story 📖

[MBL-1573](https://kickstarter.atlassian.net/browse/MBL-1573)

[MBL-1573]: https://kickstarter.atlassian.net/browse/MBL-1573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/graphql/fragments.graphql [hunk 1]
```diff
@@ -310,6 +310,11 @@ fragment reward on Reward {
     startsAt
     endsAt
     rewardType
+    allowedAddons {
```
Comment (Arkariang) [inline app/src/main/graphql/fragments.graphql:313]: Backing -> reward -> allowedAddons was never queried, therefore reward.hasAddOns was always false for the backed reward

File: app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt [hunk 2]
```diff
@@ -171,7 +178,7 @@ class AddOnsViewModel(val environment: Environment, bundle: Bundle? = null) : Vi
 
     private fun getAddOns(selectedShippingRule: ShippingRule) {
         // - Do not execute call unless reward has addOns
-        if (currentUserReward.hasAddons() || backing?.addOns().isNotNull()) {
+        if (currentUserReward.hasAddons()) {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt:181]: We want to load addOns only if the currently selected reward has them

File: app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt [hunk 3]
```diff
@@ -233,10 +241,23 @@ class GetShippingRulesUseCase(
...
         if (project.isBacking()) ShippingRule.builder()
             .apply {
+                val backing = project.backing()
                 val locationId = project.backing()?.locationId() ?: 0L
-                val locationName = project.backing()?.locationName() ?: ""
-
-                this.location(Location.Builder().id(locationId).name(locationName).displayableName(locationName).build())
+                this.id(locationId)
+                val reward = backing?.reward()?.let {
+                    if (RewardUtils.shipsToRestrictedLocations(it)) {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt:248]: for restricted locations reward will have the list of shipping rules with proper location id, name and cost

File: app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt [hunk 4]
```diff
@@ -233,10 +241,23 @@ class GetShippingRulesUseCase(
...
+                        val rule = backing?.reward()?.shippingRules()
+                            ?.first { it.location()?.id() == locationId }
+                        this.location(rule?.location())
+                        this.id(rule?.id())
+                        this.cost(rule?.cost() ?: 0.0)
+                    }
+                    if (RewardUtils.shipsWorldwide(it)) {
+                        val locationName = project.backing()?.locationName() ?: ""
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt:256]: for shipping everywhere reward will have 2 shipping rules with location, name "Earth" an "Everywhere in the world", so take the location name and cost from backing.location, backing.shipping

**Review by mtgriego — APPROVED** [review]

**Pull request merged**